lib/api/objectPutPart.js

```diff
@@ -163,6 +167,39 @@ export default function objectPutPart(authInfo, request, streamingV4Params, log,
                     return next();
                 });
         },
+        // Get any pre-existing part.
+        next => {
+            const partKey = `${uploadId}${splitter}${paddedPartNumber}`;
+            return metadata.getObjectMD(mpuBucketName, partKey, log,
+                (err, res) => {
+                    // If there is no object with the same key, continue.
+                    if (err && !err.NoSuchKey) {
+                        log.error('error getting current part (if any)', {
+                            error: err,
+                            method: 'objectPutPart::metadata.getObjectMD',
+                        });
+                        return next(err);
+                    }
+                    return next(null, res);
+                });
+        },
+        // If there is a pre-existing part, update utapi metrics and delete the
+        // pre-existing part to prevent orphaned data.
+        (res, next) => {
+            if (res) {
+                prevObjectSize = res['content-length'];
```

Comment:
As discussed in-person: `prevObjectSize` is passed later in the callback of `objectPutPart`. It is initialized as `null` at the top of the file to avoid passing the value through other functions in the waterfall. It should also be `null` if there is no pre-existing object because then Utapi will not factor in the value when updating storage utilized. 😺